src/platform/linux/wayland/event_loop.rs

```diff
@@ -193,7 +193,13 @@ impl EventsLoop {
         let static_cb = unsafe { ::std::mem::transmute(Box::new(callback) as Box<FnMut(_)>) };
         let old_cb = unsafe { self.sink.lock().unwrap().set_callback(static_cb) };
 
-        while !self.interrupted.load(::std::sync::atomic::Ordering::Relaxed) {
+        loop {
+            // If `interrupt` was called, break from the loop after resetting the flag.
+            if self.interrupted.load(::std::sync::atomic::Ordering::Relaxed) {
+                self.interrupted.store(false, ::std::sync::atomic::Ordering::Relaxed);
```

Comment:
Sure, I think at the beginning of `run_forever` makes more sense as a call to `interrupt` between two calls to `run_forever` would cause the following call to `run_forever` to skip dispatch entirely and immediately return under the current behaviour of this PR. I'll fix this in a bit.